QUARKUS-2739 - Add support for JDBCStore transaction logs (#134)
quarkus-qe/quarkus-test-plans: QUARKUS-2739.md

# QUARKUS-2739 - Add support for JDBCStore transaction logs

JIRA: https://issues.redhat.com/browse/QUARKUS-2739

Add support for `JDBC journal transaction logs`.

## Scope of the testing

> A database transaction log should records all transactions and the database modifications made by each transaction. 

Reviewer: ```suggestion
A database transaction log should record all transactions and the database modifications made by each transaction. 
```

If there is any fatal system failure, this log should be enough to recover your database back to a consistent state.

> Ensure that Narayana extension deals well with JDBC datasource transaction log and all the pending transactions (due to a crash 
> on Quarkus app) are completed when the application recover the initial state. 

Reviewer: ```suggestion
Ensure that the Narayana extension deals well with JDBC datasource transaction log and that all the pending transactions (due to a crash
on the Quarkus app) are completed when the application recovers.
```

Tests should verify:

> - All supported datasources 

Reviewer: Please list the databases here

- JVM and Native mode

> - Openshift integration (We know that Openshift will require some operator or persistent volumn in a none ephimeral node in order to test this feature, but this operator doesn´t exist at this moment. So the coverage is going to assume that all K8s / Openshift configuration is stable and mature, but maybe sometimes we could experiment some flaky test due to the ephimeral persistence unit that is under the hood)

Reviewer: Openshift => OpenShift

Operator is not planned for the initial release, there needs to be documentation describing the recovery procedure. This documentation needs to be provided by eng team. We need to verify the steps.

### Existing tests

> On Quarkus test suite there are already a Narayana module that could be reused in order to verify `JDBCStore transaction logs`.

Reviewer: ```suggestion
On Quarkus test suite there is already a Narayana module that will be reused in order to verify `JDBCStore transaction logs`.
```

### Impact on resources

> Required additional time for the test execution should be around 10 minute for native platform and 5 min for JVM. Native additional time is due to the native compilation, that is going to be done in a completly new scenario decouple of the current scenarios. Is important to us have a clean scenario because we are going to monitor internal `JBOSSTSTXTABLE` db table where all the journal is stored.

Reviewer: Is important to us have a clean scenario because we are going to monitor internal `JBOSSTSTXTABLE` db table where all the journal is stored.
 =>
It is important to have a new scenario because the test is going to monitor the `JBOSSTSTXTABLE` table where all the journal entries are stored.

> 

Reviewer: No details for OpenShift checks

## Getting familiar with the feature

Following actions were taken to ensure familiarity:
- Ensure documentation provides clear explanation on configuration options
- Ensure good user experience and simplicity of use

> 

Reviewer: Add:
` - Get familiar with JDBC store test plan and test coverage for EAP product`
` - Search JIRA for JDBC store issues tracked for EAP product to learn from previous findings`
